openapi-spec/swagger.yaml

```diff
@@ -327,6 +351,44 @@ paths:
         required: true
         schema:
           type: string
+      - name: limit
+        in: query
+        description: Number of query results on multiple pages
+        schema:
+          minimum: 1
+          type: integer
+          format: int32
+      - name: offset
+        in: query
+        description: Start position of the pagination query
+        schema:
+          minimum: 0
+          type: integer
+          format: int32
+      - name: storageType
+        in: query
+        description: Sorting keyword, based on storage type
+        schema:
+          type: string
+          enum:
+          - block
+          - file
+          - object
+      - name: sortDir
+        in: query
+        description: Ascending or descending by name
+        schema:
+          type: string
+          enum:
+          - asc
+          - des
+      - name: name
+        in: query
+        description: The pattern of profile name ([A-Za-z-0-9_-])
```

Comment:
does this support regex ? if yes, do we handle https://tools.ietf.org/html/rfc3986#section-2.2

Reply:
We do support regex. But not with all combinations.(Actually openstack/SNIA don't have regex for name but DJ has with basic regex. So we thought to add the regex for name in our API.). Please mention your thoughts/suggestions.